client/scripts/views/modals/onboarding_modal/claim_with_cli.ts

```diff
@@ -0,0 +1,166 @@
+import { Account, ChainBase } from 'models';
+import app from 'state';
+import { isU8a, isHex } from '@polkadot/util';
+import { createUserWithAddress } from 'controllers/app/login';
+import Substrate from 'controllers/chain/substrate/main';
+import CodeBlock from 'views/components/widgets/code_block';
+
+import { onboardingCWIcon, onboardingArrowLeftRightIcon, onboardingCLIIcon } from '../../components/sidebar/icons';
+import AddressSwapper from '../../components/addresses/address_swapper';
+
+import OnboardingFooterActions from './footer_actions';
+
+
+interface IOnboardingCLIAttr {
+  onNext: () => void;
+  onBack: () => void;
+  address: string;
+  accountVerifiedCallback: (account: Account<any>, onNext: (account: Account<any>) => void) => void;
+}
+
+const OnboardingCLI: m.Component<IOnboardingCLIAttr, {
+  error;
+  userProvidedSignature: string;
+  secretPhraseSaved: boolean;
+  newAddress: Account<any>;
+  isEd25519: boolean;
+  loadingNewAddress: boolean;
+}> = {
+  oninit: async (vnode) => {
+    const { address } = vnode.attrs;
+    vnode.state.error = null;
+    vnode.state.loadingNewAddress = true;
+    vnode.state.newAddress = null;
+
+    // Prevent validation on empty field
+    if (address === '') {
+      vnode.state.loadingNewAddress = false;
+      vnode.state.error = 'Address should not be empty.';
+      return;
+    }
+
+    if ((app.chain.base === ChainBase.Substrate)) {
+      if (isU8a(address) || isHex(address)) {
+        vnode.state.error = 'Address must be SS58 (e.g. 5Ew4...)';
+      }
+      try {
+        (await import('@polkadot/keyring')).decodeAddress(address);
+      } catch (err) {
+        vnode.state.error = 'Invalid address';
+      }
+    }
+    if (app.user.activeAccounts.find((acct) => acct.address === address)) {
+      vnode.state.error = 'You have already linked this address';
+    }
+
+    if (!vnode.state.error) {
+      try {
+        vnode.state.newAddress = await createUserWithAddress(AddressSwapper({
+          address,
+          currentPrefix: (app.chain as Substrate).chain.ss58Format,
+        }), vnode.state.isEd25519 ? 'ed25519' : undefined, app.activeCommunityId());
+      } catch (err) {
+        vnode.state.error = err.responseJSON ? err.responseJSON.error : 'Failed to create user.';
+      }
+    }
+
+    vnode.state.loadingNewAddress = false;
+
+    m.redraw();
+  },
+  view: (vnode) => {
+    const { accountVerifiedCallback } = vnode.attrs;
+    const { loadingNewAddress } = vnode.state;
+    let content: Vnode[] = [];
+
+    if (loadingNewAddress) {
+      content = [
+        m(Spinner, {
+          size: 'lg',
+          active: true,
+        })
+      ];
+    } else {
+      if (!vnode.state.newAddress) {
+        content = [
+          vnode.state.error && m('.error-message', vnode.state.error),
+        ];
+      } else {
+        content = [
+          m('span', 'Use the secret phrase to sign the following message:'),
+          m(CodeBlock, { clickToSelect: true }, [
```

Comment:
This is only valid for Substrate chains.

Reply:
I will check the CLI option of login flow